extensions/installed-extensions/src/index.tsx

```diff
@@ -10,54 +10,67 @@ import {
 import { extensionTypes } from "./constants";
 import { formatItem, formatOutput } from "./utils";
 import fs from "fs/promises";
-
-const exec = promisify(execCb);
+import os from "os";
+import path from "path";
+
+async function getPackageJsonFiles() {
+  try {
+    const extensionsDir = path.join(os.homedir(), ".config", "raycast", "extensions");
+    const extensions = await fs.readdir(extensionsDir);
+    const packageJsonFiles = await Promise.all(
+      extensions.map(async (extension) => {
+        const packageJsonPath = path.join(extensionsDir, extension, "package.json");
+        try {
+          await fs.access(packageJsonPath, fs.constants.F_OK);
+          return packageJsonPath;
+        } catch (e) {
+          return null;
+        }
+      }),
+    );
+    return packageJsonFiles.filter((file) => file !== null) as string[];
+  } catch (e) {
+    if (e instanceof Error) {
+      showFailureToast(e.message);
+      throw new Error(e.message);
+    }
+    throw new Error("An unknown error occurred");
+  }
+}
 
 export default function IndexCommand() {
   const preferences = getPreferenceValues<Preferences.Index>();
 
-  const [installedExtensions, setInstalledExtensions] = useState<DataType[]>([]);
+  const [installedExtensions, setInstalledExtensions] = useState<ExtensionMetadata[]>([]);
   const { isLoading, data, error } = useCachedPromise(async () => {
-    const { stdout, stderr } = await exec(
-      `find ~/.config/raycast/extensions/ -name "package.json" -exec echo -n "{}: " \\;`,
-    );
-
-    if (stderr) {
-      showFailureToast(stderr);
-      throw new Error(stderr);
-    }
-
-    const files = stdout.split(": ").filter((file) => file.trim() !== "");
-
+    const files = await getPackageJsonFiles();
     let result = await Promise.all(
       files.map(async (file) => {
         const content = await fs.readFile(file, "utf-8");
         const json = await JSON.parse(content);
 
-        const author = await json.author;
-        const owner = (await json.owner) || null;
-        const name = await json.name;
+        const author: string = json.author;
+        const owner: string | undefined = json?.owner;
+        const name: string = json.name;
+
         const link = `https://raycast.com/${owner == "null" ? author : owner}/${name}`;
         const cleanedPath = file.replace("/package.json", "");
 
         return {
           path: cleanedPath,
           name,
           author: author,
-          icon: await json.icon,
-          commands: await json.commands.length,
+          icon: json.icon,
+          commandCount: json.commands.length,
           owner,
-          title: await json.title,
+          title: json.title,
           link,
-          isOrganization: owner !== null,
-          isLocalExtension: !!cleanedPath.match(/[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}/gi),
+          isLocalExtension: !cleanedPath.match(/[\da-f]{8}-[\da-f]{4}-[\da-f]{4}-[\da-f]{4}-[\da-f]{12}/gi),
```

Comment:
A local extension is one that isn't in the store, and _doesn't_ have a UUID. The original code wrongly made all store extensions labeled as local extensions